internal/controllers/topology/cluster/desired_state.go

```diff
@@ -710,8 +710,12 @@ func computeMachineDeployment(_ context.Context, s *scope.Scope, desiredControlP
 	desiredMachineDeploymentObj.Spec.Template.Labels[clusterv1.ClusterTopologyOwnedLabel] = ""
 	desiredMachineDeploymentObj.Spec.Template.Labels[clusterv1.ClusterTopologyMachineDeploymentNameLabel] = machineDeploymentTopology.Name
 
-	// Set the desired replicas.
-	desiredMachineDeploymentObj.Spec.Replicas = machineDeploymentTopology.Replicas
+	// Set the desired replicas only if external autoscaler is not controlling the MD replica count.
+	_, ok1 := machineDeploymentTopology.Metadata.Annotations[clusterv1.ClusterAutoscalerMaxAnnotation]
+	_, ok2 := machineDeploymentTopology.Metadata.Annotations[clusterv1.ClusterAutoscalerMinAnnotation]
+	if !ok1 || !ok2 {
+		desiredMachineDeploymentObj.Spec.Replicas = machineDeploymentTopology.Replicas
+	}
```

Comment:
Overall I think that the approach described in https://github.com/kubernetes-sigs/cluster-api/issues/7293#issuecomment-1262614995 makes more sense that what is proposed in this PR, but probably we should make another round of discussion when @sbueringer is back

Getting to the proposed change
- If I got right, this is going to modify the semantics about how the API behaves for adding the autoscaler, which is most likely a breaking that requires an API bump; the approach above instead does not require this.
- The proposed implementation is going tol tie current behavior to one autoscaler implementation, which will prevent eg. usage of Karpenter (https://github.com/kubernetes-sigs/cluster-api/issues/7198)
- I'm not sure that the proposed change addresses the other concerns discussed on the related issue e.g.
  - if we create a new cluster with the autoscaler set and min replicas >1, the MS will continue to be created with replicas 1 and the autoscale won't add the missing replicas until the scheduler will require more compute resources
  - if we create a new cluster without the autoscaler and then add the autoscaler, the number of replicas will be reset to 1 no matter the current number of replicas or min/max replicas
  - other edge cases due to operations order/async controllers working on the same object